Unable to create Client

When I'm creating client without Id field set, empty Id is send to Keycloak. In result Id is not auto-generated and Client can't be accessed.

```go
	keycloakClient := gocloak.Client{
		ClientID: "test",
		Name: "test",
		Enabled: false,
	}
	return keycloak.CreateClient(token.AccessToken, "test", keycloakClient)
```

In database:

```
MariaDB [db4keycloak]> select ID, CLIENT_ID, NAME from CLIENT where NAME LIKE 'test';
+----+-----------+------+
| ID | CLIENT_ID | NAME |
+----+-----------+------+
|    | test      | test |
+----+-----------+------+
```

Thank you,
there is an error in the `ClientRepresentation` model, missing `omitempty` flag for `ID` field.
And the unit tests don't fail because the `CreateClient` tests generates an `ID`, if we remove the generation of ID then test will fail:

```
--- FAIL: TestGocloak_CreateClient (1.46s)
    client_test.go:263: RESTY 2019/05/23 09:58:04
        ---------------------- REQUEST LOG -----------------------
        POST  /auth/realms/master/protocol/openid-connect/token  HTTP/1.1
        HOST   : keycloak.k8ss:8443
        HEADERS:
                     Content-Type: application/x-www-form-urlencoded
                       User-Agent: go-resty/1.10.3 (https://github.com/go-resty/resty)
        BODY   :
        client_id=admin-cli&grant_type=password&password=pass&response_type=token&username=admin
        ----------------------------------------------------------

    client_test.go:263: RESTY 2019/05/23 09:58:05
        ---------------------- RESPONSE LOG -----------------------
        STATUS 		: 200 OK
        RECEIVED AT	: 2019-05-23T09:58:05.433637-05:00
        RESPONSE TIME	: 604.532871ms
        HEADERS:
                    Cache-Control: no-store
                       Connection: keep-alive
                   Content-Length: 1796
                     Content-Type: application/json
                             Date: Thu, 23 May 2019 14:58:05 GMT
                           Pragma: no-cache
                       Set-Cookie: KC_RESTART=; Version=1; Expires=Thu, 01-Jan-1970 00:00:10 GMT; Max-Age=0; Path=/auth/realms/master/; HttpOnly
        BODY   :
        {
           "access_token": "eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJWWW9IQ1lfSGZTUnJIcjd2Ti1YOEx2VXdtcHVfVGhXZ21INnh4Yk9FckFJIn0.eyJqdGkiOiJlMWU1NzFmOS03NTk3LTQxZGQtYmUyYS00ODZiYzVmNjdhZjkiLCJleHAiOjE1NTg2MjM1NDUsIm5iZiI6MCwiaWF0IjoxNTU4NjIzNDg1LCJpc3MiOiJodHRwczovL2tleWNsb2FrLms4c3M6ODQ0My9hdXRoL3JlYWxtcy9tYXN0ZXIiLCJzdWIiOiI0OTk3NTg3Yi0xNWQ1LTRkODgtYjgxOC1kZGJhM2ZiNGI0MjMiLCJ0eXAiOiJCZWFyZXIiLCJhenAiOiJhZG1pbi1jbGkiLCJhdXRoX3RpbWUiOjAsInNlc3Npb25fc3RhdGUiOiIxZTkyMTdiZS0xNmE0LTRlNGEtOTVhOS1mOTQwZGEwMGM1NzQiLCJhY3IiOiIxIiwic2NvcGUiOiJwcm9maWxlIGVtYWlsIiwiZW1haWxfdmVyaWZpZWQiOmZhbHNlLCJwcmVmZXJyZWRfdXNlcm5hbWUiOiJhZG1pbiJ9.qM9ojQeDylSw3LtubOWJA23QfJAvtGJe7amadqOmVajfcwtb3Q1oN-FBB-vnjvqcw_joUsv7miBNRZgiL4Oem_FZQ6T56JzTaRtCRKT29Kj1DaEI0iuGgvS4TGwf5C-KPWC62ZfpMw_wIjjR0zA856xO0eQZfUZZDwUwu033aa52T_KVLvZskUVnMqWzeRUkum33zvgtQk3LWy254fESAdB10z00GuH2l_LD1qYI1wtYldlNi8oIVBm5qIbzwp-AjBZ2xTLmehCWAglhgarjS1Gm76IoW6eYE6GNr8_Og6bT3hKi-4jBVMspp2ktyzPO4kNtGZ3TA054qW-Lm0-YJQ",
           "expires_in": 60,
           "refresh_expires_in": 1800,
           "refresh_token": "eyJhbGciOiJIUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJiMDUyZWFlMC03NzA4LTQyNTgtYjU4Yy0wNmY4ODMyN2QxMTgifQ.eyJqdGkiOiJjODkxNTA0YS02NmY3LTRiYTktODEwNi0wNGI4ODQ0NDgyYmIiLCJleHAiOjE1NTg2MjUyODUsIm5iZiI6MCwiaWF0IjoxNTU4NjIzNDg1LCJpc3MiOiJodHRwczovL2tleWNsb2FrLms4c3M6ODQ0My9hdXRoL3JlYWxtcy9tYXN0ZXIiLCJhdWQiOiJodHRwczovL2tleWNsb2FrLms4c3M6ODQ0My9hdXRoL3JlYWxtcy9tYXN0ZXIiLCJzdWIiOiI0OTk3NTg3Yi0xNWQ1LTRkODgtYjgxOC1kZGJhM2ZiNGI0MjMiLCJ0eXAiOiJSZWZyZXNoIiwiYXpwIjoiYWRtaW4tY2xpIiwiYXV0aF90aW1lIjowLCJzZXNzaW9uX3N0YXRlIjoiMWU5MjE3YmUtMTZhNC00ZTRhLTk1YTktZjk0MGRhMDBjNTc0Iiwic2NvcGUiOiJwcm9maWxlIGVtYWlsIn0.E7oumMychyHSRClTPzO7c8y-OuvWWrYoyAEOmvaKXOk",
           "token_type": "bearer",
           "not-before-policy": 0,
           "session_state": "1e9217be-16a4-4e4a-95a9-f940da00c574",
           "scope": "profile email"
        }
        ----------------------------------------------------------

    client_test.go:263: RESTY 2019/05/23 09:58:05
        ---------------------- REQUEST LOG -----------------------
        POST  /auth/admin/realms/gocloak/clients  HTTP/1.1
        HOST   : keycloak.k8ss:8443
        HEADERS:
                           Accept: application/json
                    Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJWWW9IQ1lfSGZTUnJIcjd2Ti1YOEx2VXdtcHVfVGhXZ21INnh4Yk9FckFJIn0.eyJqdGkiOiJlMWU1NzFmOS03NTk3LTQxZGQtYmUyYS00ODZiYzVmNjdhZjkiLCJleHAiOjE1NTg2MjM1NDUsIm5iZiI6MCwiaWF0IjoxNTU4NjIzNDg1LCJpc3MiOiJodHRwczovL2tleWNsb2FrLms4c3M6ODQ0My9hdXRoL3JlYWxtcy9tYXN0ZXIiLCJzdWIiOiI0OTk3NTg3Yi0xNWQ1LTRkODgtYjgxOC1kZGJhM2ZiNGI0MjMiLCJ0eXAiOiJCZWFyZXIiLCJhenAiOiJhZG1pbi1jbGkiLCJhdXRoX3RpbWUiOjAsInNlc3Npb25fc3RhdGUiOiIxZTkyMTdiZS0xNmE0LTRlNGEtOTVhOS1mOTQwZGEwMGM1NzQiLCJhY3IiOiIxIiwic2NvcGUiOiJwcm9maWxlIGVtYWlsIiwiZW1haWxfdmVyaWZpZWQiOmZhbHNlLCJwcmVmZXJyZWRfdXNlcm5hbWUiOiJhZG1pbiJ9.qM9ojQeDylSw3LtubOWJA23QfJAvtGJe7amadqOmVajfcwtb3Q1oN-FBB-vnjvqcw_joUsv7miBNRZgiL4Oem_FZQ6T56JzTaRtCRKT29Kj1DaEI0iuGgvS4TGwf5C-KPWC62ZfpMw_wIjjR0zA856xO0eQZfUZZDwUwu033aa52T_KVLvZskUVnMqWzeRUkum33zvgtQk3LWy254fESAdB10z00GuH2l_LD1qYI1wtYldlNi8oIVBm5qIbzwp-AjBZ2xTLmehCWAglhgarjS1Gm76IoW6eYE6GNr8_Og6bT3hKi-4jBVMspp2ktyzPO4kNtGZ3TA054qW-Lm0-YJQ
                     Content-Type: application/json
                       User-Agent: go-resty/1.10.3 (https://github.com/go-resty/resty)
        BODY   :
        {
           "clientId": "ClientID50415",
           "id": ""
        }
        ----------------------------------------------------------

    client_test.go:263: RESTY 2019/05/23 09:58:05
        ---------------------- RESPONSE LOG -----------------------
        STATUS 		: 409 Conflict
        RECEIVED AT	: 2019-05-23T09:58:05.838676-05:00
        RESPONSE TIME	: 404.585444ms
        HEADERS:
                       Connection: keep-alive
                   Content-Length: 54
                     Content-Type: application/json
                             Date: Thu, 23 May 2019 14:58:05 GMT
        BODY   :
        {
           "errorMessage": "Client ClientID50415 already exists"
        }
        ----------------------------------------------------------

    client_test.go:263: RESTY 2019/05/23 09:58:05
        ---------------------- REQUEST LOG -----------------------
        DELETE  /auth/admin/realms/gocloak/clients/  HTTP/1.1
        HOST   : keycloak.k8ss:8443
        HEADERS:
                           Accept: application/json
                    Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJWWW9IQ1lfSGZTUnJIcjd2Ti1YOEx2VXdtcHVfVGhXZ21INnh4Yk9FckFJIn0.eyJqdGkiOiJlMWU1NzFmOS03NTk3LTQxZGQtYmUyYS00ODZiYzVmNjdhZjkiLCJleHAiOjE1NTg2MjM1NDUsIm5iZiI6MCwiaWF0IjoxNTU4NjIzNDg1LCJpc3MiOiJodHRwczovL2tleWNsb2FrLms4c3M6ODQ0My9hdXRoL3JlYWxtcy9tYXN0ZXIiLCJzdWIiOiI0OTk3NTg3Yi0xNWQ1LTRkODgtYjgxOC1kZGJhM2ZiNGI0MjMiLCJ0eXAiOiJCZWFyZXIiLCJhenAiOiJhZG1pbi1jbGkiLCJhdXRoX3RpbWUiOjAsInNlc3Npb25fc3RhdGUiOiIxZTkyMTdiZS0xNmE0LTRlNGEtOTVhOS1mOTQwZGEwMGM1NzQiLCJhY3IiOiIxIiwic2NvcGUiOiJwcm9maWxlIGVtYWlsIiwiZW1haWxfdmVyaWZpZWQiOmZhbHNlLCJwcmVmZXJyZWRfdXNlcm5hbWUiOiJhZG1pbiJ9.qM9ojQeDylSw3LtubOWJA23QfJAvtGJe7amadqOmVajfcwtb3Q1oN-FBB-vnjvqcw_joUsv7miBNRZgiL4Oem_FZQ6T56JzTaRtCRKT29Kj1DaEI0iuGgvS4TGwf5C-KPWC62ZfpMw_wIjjR0zA856xO0eQZfUZZDwUwu033aa52T_KVLvZskUVnMqWzeRUkum33zvgtQk3LWy254fESAdB10z00GuH2l_LD1qYI1wtYldlNi8oIVBm5qIbzwp-AjBZ2xTLmehCWAglhgarjS1Gm76IoW6eYE6GNr8_Og6bT3hKi-4jBVMspp2ktyzPO4kNtGZ3TA054qW-Lm0-YJQ
                     Content-Type: application/json
                       User-Agent: go-resty/1.10.3 (https://github.com/go-resty/resty)
        BODY   :
        ***** NO CONTENT *****
        ----------------------------------------------------------

    client_test.go:263: RESTY 2019/05/23 09:58:06
        ---------------------- RESPONSE LOG -----------------------
        STATUS 		: 405 Method Not Allowed
        RECEIVED AT	: 2019-05-23T09:58:06.28965-05:00
        RESPONSE TIME	: 450.701411ms
        HEADERS:
                       Connection: keep-alive
                   Content-Length: 0
                             Date: Thu, 23 May 2019 14:58:05 GMT
        BODY   :

        ----------------------------------------------------------

    client_test.go:67: client_test.go:647: DeleteClient failed: 405 Method Not Allowed
FAIL
exit status 1
FAIL	github.com/Nerzal/gocloak	18.059s
```

--- a/client_test.go
+++ b/client_test.go
@@ -626,26 +626,67 @@ func TestGocloak_CreateClientScope(t *testing.T) {
 	FailIfErr(t, err, "DeleteClientScope failed")
 }
 
-func TestGocloak_CreateClient(t *testing.T) {
+func TestGocloak_CreateListGetDeleteClient(t *testing.T) {
 	t.Parallel()
 	cfg := GetConfig(t)
 	client := NewClientWithDebug(t)
 	token := GetAdminToken(t, client)
+	clientID := GetRandomName("ClientID")
+	t.Logf("Client ID: %s", clientID)
 
-	newClient := Client{
-		ID:       GetRandomName("ID"),
-		ClientID: GetRandomName("ClientID"),
-	}
+	// Creating a client
 	err := client.CreateClient(
 		token.AccessToken,
 		cfg.GoCloak.Realm,
-		newClient)
+		Client{
+			ClientID: clientID,
+		},
+	)
 	FailIfErr(t, err, "CreateClient failed")
+
+	// Looking for a created client
+	clients, err := client.GetClients(
+		token.AccessToken,
+		cfg.GoCloak.Realm,
+		GetClientsParams{
+			ClientID: clientID,
+		},
+	)
+	FailIfErr(t, err, "CreateClients failed")
+	FailIf(t, len(*clients) != 1, "GetClients should return exact 1 client")
+	t.Logf("Clients: %+v", *clients)
+
+	// Getting exact client
+	createdClient, err := client.GetClient(
+		token.AccessToken,
+		cfg.GoCloak.Realm,
+		(*clients)[0].ID,
+	)
+	FailIfErr(t, err, "GetClient failed")
+	t.Logf("Created client: %+v", createdClient)
+
+	// Checking that GetClient returns same client
+	AssertEquals(t, (*clients)[0], *createdClient)
+
+	// Deleting the client
 	err = client.DeleteClient(
 		token.AccessToken,
 		cfg.GoCloak.Realm,
-		newClient.ID)
+		createdClient.ID,
+	)
 	FailIfErr(t, err, "DeleteClient failed")
+
+	// Verifying that the client was deleted
+	clients, err = client.GetClients(
+		token.AccessToken,
+		cfg.GoCloak.Realm,
+		GetClientsParams{
+			ClientID: clientID,
+		},
+	)
+	FailIfErr(t, err, "CreateClients failed")
+	FailIf(t, len(*clients) != 0, "GetClients should not return any clients")
+
 }
 
 func TestGocloak_GetGroups(t *testing.T) {

--- a/models.go
+++ b/models.go
@@ -278,7 +278,7 @@ type Client struct {
 	Enabled                            bool                           `json:"enabled,omitempty"`
 	FrontChannelLogout                 bool                           `json:"frontChannelLogout,omitempty"`
 	FullScopeAllowed                   bool                           `json:"fullScopeAllowed,omitempty"`
-	ID                                 string                         `json:"id"`
+	ID                                 string                         `json:"id,omitempty"`
 	ImplicitFlowEnabled                bool                           `json:"implicitFlowEnabled,omitempty"`
 	Name                               string                         `json:"name,omitempty"`
 	NodeReRegistrationTimeout          int32                          `json:"nodeReRegistrationTimeout,omitempty"`

